src/main/php/lang/reflection/Constant.class.php

```diff
@@ -22,6 +22,26 @@ public function comment() { return Reflection::meta()->constantComment($this->re
   /** Returns a compound name consisting of `[CLASS]::$[NAME]`  */
   public function compoundName(): string { return strtr($this->reflect->class, '\\', '.').'::'.$this->reflect->name; }
 
+  /** @return lang.reflection.Constraint */
+  public function constraint() {
+    $present= true;
+
+    // Only use meta information if necessary
+    $api= function($set) use(&$present) {
+      $present= $set;
+      return Reflection::meta()->constantType($this->reflect);
+    };
+
+    // FIXME use `PHP_VERSION_ID >= 80300 ? $this->reflect->getType() : null`
+    // when php/php-src#10444 is merged.
+    $t= Type::resolve(
+      method_exists($this->reflect, 'getType') ? $this->reflect->getType() : null,
```

Comment:
As stated in comment, this `method_exists()` call should be turned into `PHP_VERSION_ID >= 80300` once php/php-src#10444 is merged

Reply:
✅ The PR is merged, the mentioned change has been committed in xp-framework/reflection@2088bc4, tests are green: https://github.com/xp-framework/reflection/actions/runs/4724603101